### PR TITLE
Migrate Fireworks from tracker-sub-DetId to TrackerTopology

### PR DESCRIPTION
**Pull request by pieterdavid**
This PR bundles a few different changes, all related to changing Fireworks code from using PXBDetId, TIBDetId etc. to TrackerTopology.
The solution we converged on, driven by the requirement from the Fireworks side to run the event display with as few dependencies as possible, was to save a copy of trackerParameters.xml with the geometry, and construct a TrackerTopology from it using StandaloneTrackerTopology (also used for some ROOT macros and SiStrip payload inspector plugins), which we propose to move to a separate package (it used to be in CalibTracker/SiStripCommon, which pulled in a number of unrelated cmssw dependencies), and changed to use tinyxml instead of xerces-c for the xml parsing.

Many thanks to @alja for doing the code changes in the Fireworks packages.

CC: @OlivierBondu @alesaggio @vidalm 
Tracked at https://github.com/cms-sw/cmssw/issues/19398

### CODE REVIEW

**Comment by boudoul** [comment]
thanks @pieterdavid !

**Comment by arunhep** [comment]
please test

**Comment by mmusich** [comment]
Hi @pieterdavid,
there are now unresolved conflicts with the master branch after #21265 has been merged. 
Can you please rebase this PR? 
Also, since you are adding this new package:
`CalibTracker/StandaloneTrackerTopology`
can you please create a PR for https://github.com/cms-sw/cms-bot/blob/master/categories.py to assign a category? As it is now, it should be under `alca`, but maybe the whole package is better placed somewhere else in `DataFormats` ?

**Comment by pieterdavid** [comment]
Hi @mmusich , thanks for the feedback, I submitted the pull request for https://github.com/cms-sw/cms-bot/blob/master/categories.py with the `alca` category as https://github.com/cms-sw/cms-bot/pull/917, but indeed that may not be the best - I would say `geometry` rather than `DataFormats` (because that's where the most similar cmssw code is), but I'm not familiar with the rules for assigning a category.

**Comment by alja** [comment]
+1

**Comment by ggovi** [comment]
+1

**Comment by arunhep** [comment]
+1

File: Fireworks/Core/src/FWGeometry.cc [hunk 1]
```diff
@@ -10,6 +10,10 @@
 #include "Fireworks/Core/interface/fwLog.h"
 #include "DataFormats/DetId/interface/DetId.h"
 
+
+// AMT deprication of tracker specific DetIds
+#include "CalibTracker/StandaloneTrackerTopology/interface/StandaloneTrackerTopology.h"
```
Comment (davidlange6) [inline Fireworks/Core/src/FWGeometry.cc]: hi @alja - does fireworks already depend on this package? Its naively not part of fwlite @smuzaffar

Comment (smuzaffar) [reply Fireworks/Core/src/FWGeometry.cc]: @davidlange6 , no CalibTracker/StandaloneTrackerTopology is not part of fwlite build set. If @alja is happy with changes in Fireworks/Core/BuildFile.xml then we will update fwlite application set to include CalibTracker/StandaloneTrackerTopology

**Comment by pieterdavid** [comment]
ping... @alja @smuzaffar is there any update on adding this package also to fwlite? There are some changes to various packages in this PR (renames etc.), so it may be best not to wait too long with merging it, if that's fine with everyone. Thanks in advance!

**Comment by fabiocos** [comment]
@alja Could you please reply to Shahzad so as we can move forward and close this PR?

**Comment by fabiocos** [comment]
+1

**Pull request merged**